app/src/test/java/com/krivosheev/mikhail/unittests/RepositoryImplTest.kt

```diff
@@ -0,0 +1,45 @@
+package com.krivosheev.mikhail.unittests
+
+import android.util.SparseArray
+import androidx.core.util.set
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import java.util.*
+
+class RepositoryImplTest {
+    private val randomNameGenerator: RandomNameGeneratorImpl = mockk()
+    private val cache = SparseArray<Backend.Result>()
+    private val repository = RepositoryImpl(randomNameGenerator, cache)
+
+    @Test
+    fun `should return backend_result when cache is empty`() {
+        val id = 5
```

Comment:
We can declare it in a companion object and mock different behaviour in each test